src/client.ts

```diff
@@ -435,6 +460,12 @@ export class InnerConnection {
   }
 
   async reconnect(singleAttempt: boolean = false): Promise<ConnectionImpl> {
+    if (this[DETACHED]) {
+      throw new errors.InterfaceError(
+        "Connection has been released to a pool"
```

Comment:
How is this possible without explicit `acquire()` methods in public API? Checking just in case?

Reply:
I see you remove it entirely in one of the following commits, never mind!